MagneticField/VolumeBasedEngine/src/MagGeometry.cc

```diff
@@ -118,8 +129,12 @@ MagVolume const* MagGeometry::findVolume1(const GlobalPoint& gp, double toleranc
   } else {  // Endcaps
     for (vector<MagVolume6Faces const*>::const_iterator v = theEVolumes.begin(); v != theEVolumes.end(); ++v) {
```

Comment:
```for( auto const& v : theEVolumes)```

Reply:
The code was written well before this syntax made its way in the C++ standard (even before "auto"). While I am (obviously) not unhappy if the code gets updated to the latest standard, I think that is a bit off the scope of the current PR.

Reply:
This file was only changed because it was causing crashes and excess error messages while I was debugging. I put in some safety checks, but more revision of this file is totally unrelated to DD4hep migration.